Fix reading ExtendedAddress from otbr-agent

tleacmcsa opened the pull request:
#### Summary

The property is defined to not emit signals on changes, which means the generated caching proxy will never have a value. Instead, we must directly read the value with the Get function and cache it ourselves.

#### Related issues

Fixes: #39722

#### Testing

Build and run examples/lighting-app/linux example with the --thread option and with otbr-agent running.

Before the fix, output contains (note 0000000000000000.local):

[1750947235.348] [620813:620813] [DL] Using Thread extended MAC for hostname.
[1750947235.348] [620813:620813] [DIS] Advertise commission parameter vendorID=65521 productID=32769 discriminator=3840/15 cm=1 cp=0
[1750947235.348] [620813:620813] [DIS] Responding with _matterc._udp.local
[1750947235.348] [620813:620813] [DIS] Responding with 52A63BCD348800B8._matterc._udp.local
[1750947235.348] [620813:620813] [DIS] Responding with 0000000000000000.local

After the fix, output contains:

[1750949673.937] [490770:490770] [DL] Using Thread extended MAC for hostname.
[1750949673.937] [490770:490770] [DIS] Advertise commission parameter vendorID=65521 productID=32769 discriminator=3840/15 cm=1 cp=0
[1750949673.937] [490770:490770] [DIS] Responding with _matterc._udp.local
[1750949673.937] [490770:490770] [DIS] Responding with 43DA52909547070B._matterc._udp.local
[1750949673.937] [490770:490770] [DIS] Responding with 6A58CF8656C9F369.local

and

"ot-ctl extaddr" in the above example test returns:
6a58cf8656c9f369
Done

tleacmcsa committed:
Fix reading ExtendedAddress from otbr-agent

The property is defined to not emit signals on changes, which means
the generated caching proxy will never have a value. Instead, we must
directly read the value with the Get function and cache it ourselves.

Fixes #39722

tleacmcsa committed:
Review fixes

- added missing error return
- use CHIP_ERROR_KEY_NOT_FOUND instead of CHIP_ERROR_INTERNAL, following
  _GetThreadProvision implementation.

tleacmcsa commented on src/platform/Linux/ThreadStackManagerImpl.cpp:
```diff
@@ -545,15 +545,65 @@
+            ChipLogError(DeviceLayer, "ERROR: Property 'ExtendedAddress' returned unexpected type: %s",
+                    g_variant_get_type_string(value.get()));
+        }
+     }
```
comment already says it does not change.

tleacmcsa commented on src/platform/Linux/ThreadStackManagerImpl.cpp:
```diff
@@ -545,15 +545,65 @@ CHIP_ERROR ThreadStackManagerImpl::_GetAndLogThreadTopologyFull()
+        {
+            ChipLogError(DeviceLayer, "ERROR: Property 'ExtendedAddress' returned unexpected type: %s",
+                    g_variant_get_type_string(value.get()));
+        }
```
Thanks, gemini!

tleacmcsa committed:
restyled

jmartinez-silabs commented on src/platform/Linux/ThreadStackManagerImpl.cpp:
```diff
@@ -546,14 +546,64 @@ CHIP_ERROR ThreadStackManagerImpl::_GetAndLogThreadTopologyFull()
+                                                           g_variant_new("(ss)", "io.openthread.BorderRouter", "ExtendedAddress"),
+                                                           G_DBUS_CALL_FLAGS_NONE, -1, nullptr, &err.GetReceiver()));
+
+        if (err)
```
The use of
VerifyOrReturnError(err, CHIP_ERROR_INTERNAL, DeviceLayer, "openthread: failed to read ExtendedAddress property: %s", err->message);
 
can improve readability here and for the others check

tleacmcsa replied:
Good idea, thanks.  Done.

tleacmcsa committed:
review comments

use VerifyOrReturnError for neatness.

tleacmcsa committed:
Remove unused DBusOpenthread properties

Since Linux platform uses a caching proxy to read OpenThread properties
over dbus, it will not provide values for properties annotated as
EmitsChangedSignal=false, so it is misleading and error-prone to
generate get function accessors for those. These properties were
therefore removed from generation. They werent being used anyway.

tleacmcsa committed:
Revert "Remove unused DBusOpenthread properties"

This reverts commit 923a8da01a294e82d98a7534d63f17b5a341b43a.

tleacmcsa committed:
make referenced otbr dbus props writable only

They are being written to, so the caching proxy isnt an issue
but we still need the properties in the generated interface.
